### PR TITLE
[12.0] add a configuration for default shift status on attendance sheets

### PR DESCRIPTION
**Pull request by robinkeunen**
## Description

Add a configuration for default shift status on attendance sheets

## Odoo task (if applicable)

https://gestion.coopiteasy.be/web#id=8282&view_type=form&model=project.task

## Checklist before approval

- [x] Tests are present (or not needed).
- [x] Credits/copyright have been changed correctly.
- [x] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: beesdoo_shift_attendance/models/attendance_sheet.py [hunk 1]
```diff
@@ -33,6 +33,14 @@ def pre_filled_task_type_id(self):
         task_types = self.env["beesdoo.shift.type"]
         return task_types.browse(tasktype_id)
 
+    @api.model
+    def _default_attendance_sheet_shift_sheet(self):
+        parameters = self.env["ir.config_parameter"].sudo()
+        return parameters.get_param(
+            "beesdoo_shift_attendance.attendance_sheet_default_shift_state",
+            default="absent_2",
```
Comment (carmenbianca) [inline beesdoo_shift_attendance/models/attendance_sheet.py:41]: Ought this default not be set in res_config_settings?

File: beesdoo_shift_attendance/tests/test_beesdoo_shift.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def setUp(self):
         self.pre_filled_task_type_id = (
             self.env["ir.config_parameter"]
             .sudo()
-            .get_param("beesdoo_shift.pre_filled_task_type_id")
+            .get_param("beesdoo_shift_attendance.pre_filled_task_type_id")
```
Comment (carmenbianca) [inline beesdoo_shift_attendance/tests/test_beesdoo_shift.py:23]: How come this didn't break the test before?

File: beesdoo_shift_attendance/tests/test_beesdoo_shift.py [hunk 3]
```diff
@@ -186,9 +187,7 @@ def test_attendance_sheet_creation(self):
             self.assertEqual(shift.task_type_id, shift.task_id.task_type_id)
             self.assertEqual(shift.super_coop_id, shift.task_id.super_coop_id)
             self.assertEqual(shift.working_mode, shift.task_id.working_mode)
-
-            # Status should be "absent" for all shifts
-            self.assertEquals(shift.state, "absent_2")
+            self.assertEquals(shift.state, "done")  # cf settings
```
Comment (carmenbianca) [inline beesdoo_shift_attendance/tests/test_beesdoo_shift.py]: assertEqual

**Review by carmenbianca — COMMENTED** [review]
Mostly LGTM to me. Some non-blocker comments/questions.

File: beesdoo_shift_attendance/models/attendance_sheet.py (hunk 1 above)
Comment (robinkeunen) [reply beesdoo_shift_attendance/models/attendance_sheet.py:41]: The default argument you mean ?

`res_config_settings` is a special wizard model that is used to write in `ir.config_parameter` (usually). The default on `res_config_settings` only therefore add a default value on the wizard when it is opened for the first time. It is not a system-wide default.

File: beesdoo_shift_attendance/tests/test_beesdoo_shift.py (hunk 2 above)
Comment (robinkeunen) [reply beesdoo_shift_attendance/tests/test_beesdoo_shift.py:23]: because of the aforementioned `default` arg.

**Comment by robinkeunen** [comment]
@carmenbianca thanks for the review, I replied or fixed.

**Review by remytms — APPROVED** [review]
LGTM

**Pull request merged**